packages/jupyter-ai/src/components/settings/model-parameters-input.tsx

```diff
@@ -94,53 +116,75 @@ export function ModelParametersInput(): JSX.Element {
       return;
     }
 
+    if (!props.modelId) {
+      setValidationError('No model selected');
+      return;
+    }
+
     // Filter out parameters with empty values
     const validParams = parameters.filter(param => param.value.trim() !== '');
 
+    // Validation: Check boolean values
+    const invalidBooleanParams = validParams.filter(param => {
+      if (param.type.toLowerCase() === 'boolean') {
+        const value = param.value.toLowerCase().trim();
+        return value !== 'true' && value !== 'false';
+      }
+      return false;
```

Comment:
Note that this will only work if users type `boolean` and not `bool` or something else, which might not be evident as a requirement / expectation. Ideally we'd provide dropdowns for types and and also for values that have a small set of allowed values. We discussed that perfecting UX is out of scope of this PR so I just wanted to put a note and bring everyone's attention to this.

Reply:
Thanks Andrii. I've opened #1461 to track this work.